classes/blocks/class-base-block.php

```diff
@@ -94,4 +96,93 @@ protected function is_rest_request() {
 	public static function update_data( array $fields ) {
 		throw new NotImplemented( 'Method update_data is not implemented for ' . static::class );
 	}
+
+	/**
+	 * Register scripts and styles for a block
+	 */
+	public static function enqueue_editor_assets() {
+		$camelized_block_name = static::get_camelized_block_name();
+		$url_path             = trailingslashit( P4GBKS_PLUGIN_URL ) . 'assets/build/' . $camelized_block_name;
+		$dir_path             = trailingslashit( P4GBKS_PLUGIN_DIR ) . 'assets/build/' . $camelized_block_name;
+		$block_name           = static::get_full_block_name();
+
+		// Editor Script.
+		wp_enqueue_script(
+			$block_name . '-editor-script',
```

Comment:
Do we need to enqueue this ourselves for the editor? I think it can also be passed with `register_block_type`.

Reply:
It's documented here, but only in the comments :roll_eyes: https://developer.wordpress.org/reference/functions/register_block_type/#user-contributed-notes

Reply:
Yes, I enqueued them in `register_block_type` initially, along with the frontend scripts, but as it always enqueues the frontend scripts no matter if the block is present or not, I moved the frontend scripts apart, and then I moved the editor scripts apart too because I found it weird to have some enqueued by `register_block_type` and some outside it.

Reply:
It may be a bit weird but it's a more high level way of writing it since it avoids the enqueue logic, we just pass a name. Also if WP decides to lazy load the scripts in the editor (not sure if they do or plan to), then using their API will allow us to benefit from that.

Reply:
Hmmm not sure about this, we will still need to keep the frontend styles apart from `register_block_type`. So it would mean we'll have to keep `enqueue_editor_assets` anyway... OR, maybe we can include the frontend styles in the editor styles build. What do you think?

Reply:
> maybe we can include the frontend styles in the editor styles build

That may be a simple solution too, probably all individual scripts are always loaded in the editor anyway, so performance would be about the same. And it's not very performance senstive too.

Reply:
Circling back to this: as mentioned in another comment, since the WP implementation of the handling of these files changed quite a bit, I would keep this enqueueing logic in our side, so we have more control over it and it's consistent in both sides (editor & frontend). As you said, we don't know what plans WP has for them. I think it would be better to catch up with the WP implementation later, using the JSON metadata files and such.